app/src/main/java/edu/usf/sas/pal/muser/util/HeadphoneUtils.kt

```diff
@@ -0,0 +1,111 @@
+package edu.usf.sas.pal.muser.util
+
+import android.content.Context
+import android.media.AudioDeviceInfo
+import android.media.AudioManager
+import android.os.Build
+import android.util.Log
+import edu.usf.sas.pal.muser.model.AudioData
+import edu.usf.sas.pal.muser.model.AudioDeviceType
+import edu.usf.sas.pal.muser.model.VolumeData
+
+object HeadphoneUtils {
```

Comment:
Actually, looks like this class holds a state in `deviceType` - typically utility classes don't hold states, they are entirely static.

Is the only reason there is a `deviceType` within the class because you can't return two values from `isSpeakerOn()`? If so I'd prefer you eliminate the class variable and return a `Pair` (holding both the Boolean and deviceType) instead - https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-pair/.